core/app/models/spree/order/checkout.rb

```diff
@@ -291,6 +297,12 @@ def assign_default_credit_card
             end
           end
 
+          def handle_failed_payments
```

Comment:
This method looks a bit confusing to me. Can you rename `errors` to maybe `base_errors`? This way you're not forced to call `self`. :)